src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -45,5 +48,1659 @@ class Outer
+        {
+            var source1 = """
+                using System;
+
+                class Outer
+                {
+                    file class C
+                    {
+                        public static void M()
+                        {
+                            Console.Write(1);
+                        }
+                    }
+                }
+                """;
+
+            var source2 = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        Outer.C.M(); // 1
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(new[] { source1, source2 });
+            comp.VerifyDiagnostics(
+                // (5,15): error CS0117: 'Outer' does not contain a definition for 'C'
+                //         Outer.C.M(); // 1
+                Diagnostic(ErrorCode.ERR_NoSuchMember, "C").WithArguments("Outer", "C").WithLocation(5, 15));
+        }
+
+        [Fact]
+        public void AccessThroughGlobalUsing_01()
+        {
+            var usings = """
+                global using NS;
+                """;
+
+            var source = """
+                using System;
+
+                namespace NS
+                {
+                    file class C
+                    {
+                        public static void M() => Console.Write(1);
+                    }
+                }
+
+                class Program
+                {
+                    public static void Main()
+                    {
+                        C.M();
+                    }
+                }
+                """;
+
+            var verifier = CompileAndVerify(new[] { usings, source, IsExternalInitTypeDefinition }, expectedOutput: "1");
+            verifier.VerifyDiagnostics();
+        }
+
+        [Theory]
+        [InlineData("file ")]
+        [InlineData("")]
+        public void AccessThroughGlobalUsing_02(string fileModifier)
+        {
+            var source = $$"""
+                using System;
+
+                namespace NS
+                {
+                    {{fileModifier}}class C
+                    {
+                        public static void M() => Console.Write(1);
+                    }
+                }
+
+                class Program
+                {
+                    public static void Main()
+                    {
+                        C.M(); // 1
+                    }
+                }
+                """;
+
+            var compilation = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, options: TestOptions.DebugExe.WithUsings("NS"));
+            compilation.VerifyDiagnostics(
+                // (15,9): error CS0103: The name 'C' does not exist in the current context
+                //         C.M(); // 1
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "C").WithArguments("C").WithLocation(15, 9));
```

Comment:
This error seems unexpected. The language says members of namespace `NS` are in scope and `C` is visible within the file.

Reply:
This error occurs in the same way whether the `file` modifier is used or not. Does that mean there's an existing bug here with usings provided via compilation options?

Reply:
Feels like a bug.
Could you test with global using specified in a different file? My assumption was that compilation options and global usings would work the same.

Reply:
Filed bug #61502. I'd like to address it separately from this feature implementation if that's alright.